src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableLSOperation.java

```diff
@@ -0,0 +1,136 @@
+        System.arraycopy(encodeObUniVersionHeader(getVersion(), getPayloadContentSize()), 0, bytes,
+                idx, headerLen);
+        idx += headerLen;
+
+        // 1. encode ls id
+        System.arraycopy(Serialization.encodeI64(lsId), 0, bytes, idx, 8);
+        idx += 8;
+
+        // 2. encode option flag
+        int len = Serialization.getNeedBytes(optionFlag.getValue());
+        System.arraycopy(Serialization.encodeVi64(optionFlag.getValue()), 0, bytes, idx, len);
+        idx += len;
+
+        // 3. encode Operation
+        len = Serialization.getNeedBytes(tabletOperations.size());
+        System.arraycopy(Serialization.encodeVi64(tabletOperations.size()), 0, bytes, idx, len);
+        idx += len;
+        for (ObTableTabletOp tabletOperation : tabletOperations) {
+            len = (int) tabletOperation.getPayloadSize();
+            System.arraycopy(tabletOperation.encode(), 0, bytes, idx, len);
+            idx += len;
+        }
+
+        return bytes;
+    }
+
+    /*
+     * Decode.
+     */
+    @Override
+    public Object decode(ByteBuf buf) {
+        // 0. decode header
+        super.decode(buf);
+
+        // 1. decode others
+        this.lsId = Serialization.decodeI64(buf);
+
+        // 2. decode flags
+        long flagValue = Serialization.decodeVi64(buf);
+        optionFlag.setValue(flagValue);
+
+
+        // 3. decode Operation
+        int len = (int) Serialization.decodeVi64(buf);
+        tabletOperations = new ArrayList<ObTableTabletOp>(len);
+        for (int i = 0; i < len; i++) {
+            ObTableTabletOp tabletOperation = new ObTableTabletOp();
+            tabletOperation.decode(buf);
+            tabletOperations.add(tabletOperation);
+        }
+
+        return this;
+    }
+
+    /*
+     * Get payload content size.
+     */
+    @Override
+    public long getPayloadContentSize() {
+        long payloadContentSize = 0;
+        payloadContentSize += Serialization.getNeedBytes(tabletOperations.size());
+        for (ObTableTabletOp operation : tabletOperations) {
+            payloadContentSize += operation.getPayloadSize();
+        }
+
+        return payloadContentSize + LS_ID_SIZE + Serialization.getNeedBytes(optionFlag.getValue());
+    }
+
+    /*
+     * Get table operations.
+     */
+    public List<ObTableTabletOp> getTabletOperations() {
+        return tabletOperations;
+    }
+
+    /*
+     * Add table operation.
+     */
+    public void addTabletOperation(ObTableTabletOp tabletOperation) {
+        this.tabletOperations.add(tabletOperation);
+        int length = this.tabletOperations.size();
+        if (isSameType() && length > 1
+                && tabletOperation.isSameType()
+                && (this.tabletOperations.get(length - 1).getSingleOperations().get(0).getSingleOpType() ==
+                    this.tabletOperations.get(length - 2).getSingleOperations().get(0).getSingleOpType())) {
+            setIsSameType(false);
```

Comment:
Why set false when the operation is the same

Reply:
fix